src/core/js/models/adaptModel.js

```diff
@@ -212,6 +212,30 @@ define(function (require) {
             return parent;
         },
 
+        getParents: function() {
+            var context = this;
+            var parents = [];
+            while(true) {
+                if (!context.get("_parentId")) break;
+                context = context.getParent();
+                parents.push(context);
+            }
+            if (parents.length === 0) return null;
+            return new Backbone.Collection(parents);
+        },
+
+        getPath: function() {
+            var context = this;
+            var parents = [context];
+            while(true) {
+                if (!context.get("_parentId")) break;
+                context = context.getParent();
+                parents.push(context);
+            }
+            if (parents.length === 0) return null;
+            return new Backbone.Collection(parents);
+        },
+
```

Comment:
Suggest merging the two functions for readability producing something like this:
```js
getParents: function(shouldIncludeChild) {
    var parents = [];
    var context = this;

    if (shouldIncludeChild) parents.push(context);

    while (context.has("_parentId")) {
        context = context.getParent();
        parents.push(context);
    }

    return parents.length ? new Backbone.Collection(parents) : null;
}
```